bindings/src/main/java/io/pravega/storage/filesystem/FileSystemChunkStorage.java

```diff
@@ -0,0 +1,431 @@
+import lombok.val;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFileAttributes;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+
+/**
+ * {@link ChunkStorage} for file system based storage.
+ *
+ * Each Chunk is represented as a single file on the underlying storage.
+ * The concat operation is implemented as append.
+ */
+
+@Slf4j
+public class FileSystemChunkStorage extends BaseChunkStorage {
+    //region members
+
+    private final FileSystemStorageConfig config;
+
+    //endregion
+
+    //region constructor
+
+    /**
+     * Creates a new instance of the FileSystemChunkStorage class.
+     *
+     * @param config The configuration to use.
+     */
+    public FileSystemChunkStorage(FileSystemStorageConfig config) {
+        this.config = Preconditions.checkNotNull(config, "config");
+    }
+
+    //endregion
+
+    //region capabilities
+
+    /**
+     * Gets a value indicating whether this Storage implementation supports merge operation either natively or through appends.
+     *
+     * @return True or false.
+     */
+    @Override
+    public boolean supportsConcat() {
+        return true;
+    }
+
+    /**
+     * Gets a value indicating whether this Storage implementation supports append operation on underlying storage object.
+     *
+     * @return True or false.
+     */
+    @Override
+    public boolean supportsAppend() {
+        return true;
+    }
+
+    /**
+     * Gets a value indicating whether this Storage implementation supports truncate operation on underlying storage object.
+     *
+     * @return True or false.
+     */
+    @Override
+    public boolean supportsTruncation() {
+        return false;
+    }
+
+    //endregion
+
+    //region
+
+    @VisibleForTesting
+    protected FileChannel getFileChannel(Path path, StandardOpenOption openOption) throws IOException {
+        try {
+            return FileChannel.open(path, openOption);
+        } catch (NoSuchFileException e) {
+            throw new FileNotFoundException(path.toString());
+        }
+    }
+
+    @VisibleForTesting
+    protected long getFileSize(Path path) throws IOException {
+        try {
+            return Files.size(path);
+        } catch (NoSuchFileException e) {
+            throw new FileNotFoundException(path.toString());
+        }
+    }
+
+    /**
+     * Retrieves the ChunkInfo for given name.
```

Comment:
In general, the value I see in javadocs for overridden methods is the additional information about the specifics of the implementation. Otherwise, just copying from parent class or interface doesn't add much. I'm in favor of adding comments to reflect what is relevant for the particular implementation.

Reply:
removed redundant comments.